elliott/elliottlib/cli/find_builds_cli.py

```diff
@@ -686,24 +692,75 @@ def _filter_out_attached_builds(
     return unattached_builds, attached_to_advisories
 
 
-async def find_builds_konflux(runtime, payload):
+async def find_builds_konflux(runtime):
     """
-    Find konflux builds for group/assembly
+    Find Konflux builds for a group/assembly, separating payload and non-payload images,
+    and fetch related OLM bundle builds.
+
+    This function:
+    - Iterates over image metadata from the runtime, filtering out base-only and non-release images.
+    - For each image, determines if it is a payload image and collects its build record.
+    - Separates the results into payload and non-payload image builds.
+    - For OLM operator images, fetches the related bundle build records from the database.
+    - Returns four lists:
+        1. Build records for payload images.
+        2. Build records for non-payload images.
+        3. OLM bundle build records found.
+        4. OLM operator build records for which no bundle build was found.
+
+    Args:
+        runtime: The runtime object providing access to image metadata and the Konflux database.
+        payload: Boolean flag indicating whether to process payload images.
+
+    Returns:
+        Tuple of four lists:
+            - List of build records for payload images.
+            - List of build records for non-payload images.
+            - List of OLM bundle build records found.
+            - List of OLM operator build records for which no bundle build was found.
+
+    Raises:
+        ElliottFatalError: If any image build records could not be found.
     """
-
     runtime.konflux_db.bind(KonfluxBuildRecord)
 
-    image_metas: List[ImageMetadata] = []
+    image_metas: list[tuple[bool, ImageMetadata]] = []
     for image in runtime.image_metas():
         if image.base_only or not image.is_release:
             continue
-        if (payload and not image.is_payload) or (not payload and image.is_payload):
-            continue
-        image_metas.append(image)
+        image_metas.append((image.is_payload, image))
 
     LOGGER.info("Fetching NVRs from DB...")
-    tasks = [image.get_latest_build(el_target=image.branch_el_target()) for image in image_metas]
-    records: List[Dict] = [r for r in await asyncio.gather(*tasks) if r is not None]
-    if len(records) != len(image_metas):
-        raise ElliottFatalError(f"Failed to find Konflux builds for {len(image_metas) - len(records)} images")
-    return records
+    # find build result (is_olm_operator, build_Record, is_payload)
+    tasks = [
+        (image.is_olm_operator, image.get_latest_build(el_target=image.branch_el_target()), is_payload)
+        for is_payload, image in image_metas
+    ]
+    results = await asyncio.gather(*[task[1] for task in tasks])
```

Comment:
we're mixing things here .. 
let us not insert additional things inside task list. let's use `task` only for collecting coroutines, and something else to combine all of them